### PR TITLE
fix: update throttle

### PR DESCRIPTION
**Pull request by alexjoverm**
We've been reported in https://github.com/storyblok/storyblok-js-client/issues/682, and thanks to the discovery of @dojscart in https://github.com/withastro/adapters/issues/43#issuecomment-1782206087 (really really thanks 🙏 ) we've managed to identify that the internal queue of the throttle promise wasn't being awaited properly, causing some edge cases to fail (like Cloudflare, and likely the SSG generation).

On the way, I also managed to find another case -> after aborting the queue, promises kept being added and got stuck in the queue unresolved forever.

Solution
Basically, now the promises in the queue are awaited properly after crossing the limit, and no new promises are getting added after aborting the queue.

Fixes https://github.com/storyblok/storyblok-js-client/issues/682

How to test the issue
Run the new test npx vitest run src/throttlePromise.test.ts against the src/throttlePromise.ts living in the main branch -> both test cases should fail
Now, run again the same test against the updated src/throttlePromise.ts from this branch -> both test cases should pass correctly
Aside from that, I can recommend to try to break the expected functionality of the throttlePromise and its internal queue. I tried via a local script to heavily load the queue with all kind of scenarios - thousands of calls, randomly make them fail, etc - and so far it worked out, but extra eyes and hands are always great.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -68,7 +60,7 @@ class Storyblok {
   private client: SbFetch;
   private maxRetries: number;
   private retriesDelay: number;
-  private throttle: ThrottleFn;
+  private throttle;
```
Comment (edodusi) [inline src/index.ts:63]: Given this is assigned the return value of `throttledQueue()` on line 143, I think we can type this as following:

```suggestion
  private throttle: ReturnType<typeof throttledQueue>;
```

File: src/throttlePromise.ts [hunk 2]
```diff
@@ -1,73 +1,76 @@
...
+  }
 }
 
-interface Queue {
-  resolve: (args: any) => any;
-  reject: (args: any) => any;
-  args: any[];
-  self: any;
-}
+function isFinite(value: number) {
```
Comment (edodusi) [inline src/throttlePromise.ts:10]: do you find any specific reason why we should keep this function and not using `Number.isFinite()` instead? I guess the only reason would be type coercion but here we are using it only with `throttledQueue` and the type is already coerced

Comment (alexjoverm) [reply src/throttlePromise.ts:10]: Great point. It was already there, but good time to remove it 😉

File: src/index.ts (hunk 1 above)
Comment (alexjoverm) [reply src/index.ts:63]: 💯

**Review by edodusi — APPROVED** [review]
All good to me 👍

**Pull request merged**